### PR TITLE
Feat(eos_cli_config_gen): Add support for snmp-server ipmib ifspeed shape-rate

### LINKED ISSUE #4377: Feat (eos_cli_config_gen): Add support for SNMP server shape-rate
### Enhancement summary

Add support for the following CLI syntax (at the global config level):

`snmp-server ifmib ifspeed shape-rate`

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Enable global support for SNMP ifspeed reflecting shaping rate

### Describe the solution you would like

Add support for eos_cli_config_gen along the lines of:

```
snmp_server:
  ifmib:
     ifspeed: <str; "shape-rate">

```       

### Describe alternatives you have considered

n/a

### Additional context

n/a

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gusmb**
## Change Summary

Add support for SNMP server shape-rate

## Related Issue(s)

Fixes #4377 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Adds following knob to `snmp-server`:

```yaml
snmp_server:
  ifmib_ifspeed_shape_rate: <bool>
```

## How to test

See molecule test

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ClausHolbechArista — COMMENTED** [review]
LGTM so far. Only needs a run of molecule to get the test artifacts updated. 
```cli
cd ansible_collections/arista/avd
molecule converge -s eos_cli_config_gen -- --limit snmp
```

**Comment by ClausHolbechArista** [comment]
I have committed a few changes to your PR to unblock the CI.

**Comment by gusmb** [comment]
> cd ansible_collections/arista/avd
> molecule converge -s eos_cli_config_gen -- --limit snmp

Is it possible for you to run it? I am using the dev container at the moment and getting a few issues with pre-commit, dependencies are not installed etc. Probably need to rebuild my environment

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/snmp_server.schema.yml [hunk 1]
```diff
@@ -262,3 +262,6 @@ keys:
                 - int
             enable:
               type: bool
+      ifmib_ifspeed_shape_rate:
+        type: bool
+        description: SNMP ifspeed reflecting shaping rate
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/snmp_server.schema.yml]:
```suggestion
        description: SNMP ifspeed reflecting shaping rate.
```

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Verified command order on EOS.

**Review by MaheshGSLAB — APPROVED** [review]

**Pull request merged**